src/helpers/ValueHelpers.ts

```diff
@@ -0,0 +1,545 @@
+        };
+    },
+    /**
+     * Wraps either a string, number or bigint in the underlying signed 64-bit integer representation
+     *
+     * _ATTENTION_: plain JavaScript numbers do not suffer from a loss in precision only within the -2^53 to 2^53 range
+     *
+     * Example:
+     *
+     * ```
+     * daml.int64(99898)
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'int64',
+     *     int64: '99898'
+     * }
+     * ```
+     */
+    int64: (value: string | number | bigint): Int64Value => {
+        return {
+            valueType: "int64",
+            int64: '' + value
+        };
+    },
+    /**
+     * Wraps either a string, number or bigint in the underlying arbitrary precision decimal representation
+     *
+     * _ATTENTION_: plain JavaScript numbers do not suffer from a loss in precision only within the -2^53 to 2^53 range
+     *
+     * Example:
+     *
+     * ```
+     * daml.decimal(99898)
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'decimal',
+     *     decimal: '99898'
+     * }
+     * ```
+     */
+    decimal: (value: string | number | bigint): DecimalValue => {
+        return {
+            valueType: "decimal",
+            decimal: '' + value
+        };
+    },
+    /**
+     * Wraps a string into the underlying representation of text.
+     *
+     * Example:
+     *
+     * ```
+     * daml.text('hello, world')
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'text',
+     *     text: 'hello, world'
+     * }
+     * ```
+     */
+    text: (value: string): TextValue => {
+        return {
+            valueType: "text",
+            text: value
+        };
+    },
+    /**
+     * Wraps a string into the underlying representation of parties.
+     *
+     * Example:
+     *
+     * ```
+     * daml.party('Alice')
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'party',
+     *     party: 'Alice'
+     * }
+     * ```
+     */
+    party: (value: string): PartyValue => {
+        return {
+            valueType: "party",
+            party: value
+        };
+    },
+    /**
+     * Wraps a plain boolean into the underlying representation of boolean values on the ledger
+     *
+     * Example:
+     *
+     * ```
+     * daml.boolean(true)
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'bool',
+     *     bool: true
+     * }
+     * ```
+     */
+    bool: (value: boolean): BoolValue => {
+        return {
+            valueType: "bool",
+            bool: value
+        };
+    },
+    /**
+     * If given a {@link Date}, wraps its value into the underlying representation for timestamp.
+     *
+     * If given a number or bigint, casts it to the corresponding string value.
+     *
+     * If given a string, passes through unchanged. Pay extra caution when using this approach.
```

Comment:
I think this is too risky. People might just assume that the Ledger API gets values as ISO 9601 and they'd get a very confusing error.

I'd much rather check that the string is a valid number, and crash otherwise.

Reply:
In general I agree and I'd be happy to fix this. The reason I'm not entirely convinced is that this would mean making some validation regarding semantics on the bindings, whereas so far we exclusively checked for types and let the server handle the validation. If the server changes what it regards as valid, we would have to change it as well on the bindings, potentially introducing a breaking change. I would prefer not introducing this unless there is strong feedback that this is needed.

Even if that would be the case, I'd rather demand this check to the validation layer rather then introducing it here to not spread the validation logic across several components. Remember that the plan is also to make the validation logic pluggable, so users that could prefer a stricter validator could implement their own.

Reply:
Thought about this: what I would do to cover this case in a consistent manner is introduce validation rules in this regard only for types that have been forcefully mapped to strings due to precision issues: `int64`, `timestamp` and `date`.

I would introduce a new possible error to validation (e.g.: `invalid-numeric-representation`) and signal this error via the usual callback mechanism, rather then throwing. This way the user also has to care about a single possible source of errors instead of having to sprinkle his code with `try`s while also having to take care of errors from callbacks.

I would do this as a separate ticket and treat it as a separate concern.

Reply:
ok, sounds good.

Reply:
Working on it, tracked [here](https://github.com/digital-asset/daml-js/issues/57).